test/extensions/filters/http/jwt_authn/jwks_async_fetcher_test.cc

```diff
@@ -241,6 +244,71 @@ TEST_P(JwksAsyncFetcherTest, TestNetworkFailureFetchAndRefresh) {
   EXPECT_EQ(2U, stats_.jwks_fetch_failed_.value());
 }
 
+TEST_P(JwksAsyncFetcherTest, TestNetworkFailureFetchAndRetrySuccessfully) {
+  const char config[] = R"(
+      http_uri:
+        uri: https://pubkey_server/pubkey_path
+        cluster: pubkey_cluster
+
+      async_fetch: {}
+
+      retry_policy:
+        retry_back_off:
+          base_interval: 0.001s
```

Comment:
Are we using the real timer?   It may have race condition.  I think the mockDispatcher has SimulatedTimer which we can call advanceTimeAndRun() to call timer callback function

Reply:
I will fix these test only to use a mock dispatcher. That makes sense.

Reply:
re-wrote the two new "retrying" tests. new test class, mocking the http async client to simulate N remote fetch failures at that level, verify that the retries are indeed done. using a mock dispatcher and timer.